internal/functions/serve/templates/main.ts

```diff
@@ -69,120 +98,129 @@ async function verifyJWT(jwt: string): Promise<boolean> {
+
+        if (!isValidJWT) {
+          return getResponse({ msg: "Invalid JWT" }, STATUS_CODE.Unauthorized);
+        }
+      } catch (e) {
+        console.error(e);
+        return getResponse({ msg: e.toString() }, STATUS_CODE.Unauthorized);
       }
-    } catch (e) {
-      console.error(e);
-      return new Response(
-        JSON.stringify({ msg: e.toString() }),
-        { status: 401, headers: { "Content-Type": "application/json" } },
-      );
     }
-  }
 
-  const servicePath = `${FUNCTIONS_PATH}/${functionName}`;
-  console.error(`serving the request with ${servicePath}`);
+    const servicePath = `${FUNCTIONS_PATH}/${functionName}`;
+    console.error(`serving the request with ${servicePath}`);
 
-  const memoryLimitMb = 150;
-  const workerTimeoutMs = 400 * 1000;
-  const noModuleCache = false;
-  const envVarsObj = Deno.env.toObject();
-  const envVars = Object.entries(envVarsObj)
-    .filter(([name, _]) =>
-      !EXCLUDED_ENVS.includes(name) && !name.startsWith("SUPABASE_INTERNAL_")
-    );
-  const forceCreate = true;
-  const customModuleRoot = ""; // empty string to allow any local path
-  const cpuTimeSoftLimitMs = 1000;
-  const cpuTimeHardLimitMs = 2000;
-  try {
-    const worker = await EdgeRuntime.userWorkers.create({
-      servicePath,
-      memoryLimitMb,
-      workerTimeoutMs,
-      noModuleCache,
-      importMapPath: functionsConfig[functionName].importMapPath,
-      envVars,
-      forceCreate,
-      customModuleRoot,
-      cpuTimeSoftLimitMs,
-      cpuTimeHardLimitMs,
-    });
-    const controller = new AbortController();
-
-    const { signal } = controller;
-    // Note: Requests are aborted after 200s (same config as in production)
-    // TODO: make this configuarable
-    setTimeout(() => controller.abort(), 200 * 1000);
-    return await worker.fetch(req, { signal });
-  } catch (e) {
-    console.error(e);
-    if (e.name === WorkerErrors.InvalidWorkerCreation) {
-      return respondWith(
-        {
-          code: "BOOT_ERROR",
-          message: "Worker failed to boot (please check logs)",
-        },
-        503,
+    const memoryLimitMb = 150;
+    const workerTimeoutMs = 400 * 1000;
+    const noModuleCache = false;
+    const envVarsObj = Deno.env.toObject();
+    const envVars = Object.entries(envVarsObj)
+      .filter(([name, _]) =>
+        !EXCLUDED_ENVS.includes(name) && !name.startsWith("SUPABASE_INTERNAL_")
       );
-    }
-    if (e.name === WorkerErrors.InvalidWorkerResponse) {
-      return respondWith(
+
+    const forceCreate = true;
+    const customModuleRoot = ""; // empty string to allow any local path
+    const cpuTimeSoftLimitMs = 1000;
+    const cpuTimeHardLimitMs = 2000;
+
+    // NOTE(Nyannyacha): Decorator type has been set to tc39 by Lakshan's request,
+    // but in my opinion, we should probably expose this to customers at some
+    // point, as their migration process will not be easy.
+    const decoratorType = "tc39";
```

Comment:
yep, let's keep it `tc39` for now and can allow users' to configure it on `config.toml` (not many users use decorators on edge functions yet)